otter/convergence/steps.py

```diff
@@ -539,3 +552,91 @@ def as_effect(self):
         """
         result = StepResult.LIMITED_RETRY if self.limited else StepResult.RETRY
         return Effect(Constant((result, list(self.reasons))))
+
+
+# ----- Cloud Orchestration Steps -----
+
+
+@implementer(IStep)
+@attr.s
+class CreateStack(object):
+    """
+    A stack must be created.
+
+    :ivar pmap stack_config: Heat launch configuration.
+
+    """
+    stack_config = attr.ib(validator=instance_of(PMap))
+
+    def as_effect(self, append_stack_uuid=append_stack_uuid):
```

Comment:
oh, hold on, I didn't notice the uuid4 before. lemme think about this a bit more.